### DISCUSSION TITLE
Cleanest way to use measured POA?

### DISCUSSION THREAD

**Discussion by williamhobbs** [williamhobbs/2024_pvpmc_self_shade#5] [Q&A]
I would like to have the option to use ground-measured POA as a "stretch goal" for this work, with and without the option to supplement with satellite data. That leads me to two questions:
1. What to do about beam irradiance and diffuse fraction?
2. Where to pass the measured POA in?

**Comment by williamhobbs** [comment]
For q1:
- _Without_ supplementary satellite data, I think it's best to use `pvlib.irradiance.gti_dirint`. 
- With supplementary satellite data, I think it makes sense to at least try calculating diffuse fraction from the sat data, but based on https://www.nrel.gov/docs/fy22osti/82076.pdf, it may not help much (although I think that was an orientation-agnostic diffuse fraction, not POA diffuse fraction...)

For q2:
I think it makes sense to substitute measured POA for _poa_front_total_without_direct_shade_ here:

```python
    # shaded fraction for the whole array (all courses/strings in a row)
    fs_array = irrad_inf_sh['shaded_fraction_front']

    # work backwards to unshaded direct irradiance for the whole array:
    poa_front_direct_unshaded = irrad_inf_sh['poa_front_direct'] / (1-fs_array)

    # total poa on the front, but without direct shade impacts (keeping diffuse impacts from infinite_sheds)
    poa_front_total_without_direct_shade = irrad_inf_sh['poa_front_diffuse'] + poa_front_direct_unshaded
```

**Reply by williamhobbs** [reply]
And it might be useful to include this about considering what diffuse irradiance is or is not included in measured POA:

![image](https://github.com/williamhobbs/2024_pvpmc_self_shade/assets/45701090/d370f6fa-0050-4d0c-a6f2-481d64db826d)
![image](https://github.com/williamhobbs/2024_pvpmc_self_shade/assets/45701090/484f88e3-db31-4edd-8979-fdf80a007109)

**Reply by kandersolar** [reply]
For how to use pyranometer data in general, I'm not sure it can always slot perfectly into the modeling framework we currently have.  For example what about true-tracking systems where the pyranometer itself experiences morning/evening beam shading in certain times of year?  Even if those times can be reliably identified, I don't see how we can get to a good estimate of beam irradiance from the available data (without using supplementary data, e.g. satellite).

> I think it's best to use pvlib.irradiance.gti_dirint

An alternative that claims better convergence recently became available: https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.irradiance.ghi_from_poa_driesse_2023.html

**Reply by williamhobbs** [reply]
Good to know about `ghi_from_poa_driesse_2023`. Would you then use something like `erbs` or `erbs_driesse` to estimate DHI/DNI from the output GHI? 

Site measured GHI could also be an option that I wasn't thinking about, skipping the ghi-from-poa step...

And yes, lots of issues with pyranometer data, without even getting into mounting, calibration, cleaning, etc... Do you think _poa_front_total_without_direct_shade_ is the best (but still imperfect) place to sub it in here?

**Reply by kandersolar** [reply]
Yep, it has to be followed by a decomposition model.  I remember Anton being hesitant for the function to return DHI/DNI directly for some reason.  

And yeah, `poa_front_total_without_direct_shade` makes sense to me!
